### PR TITLE
th/build_meson_on_travis

### PR DESCRIPTION
**Pull request by thom311**
Some fixes for meson build, and enable meson build on travis.

Please review

### CODE REVIEW

**Comment by thom311** [comment]
various options are still disabled in meson build (in fact, even with autotools, we don't build everything in our CI).

Of course, we should test/build as much as possible... TODO laster.

**Comment by bengal** [comment]
LGTM

File: meson_options.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ option('bluez5_dun', type: 'boolean', value: false, description: 'enable Bluez5
 
 # configuration plugins
 option('config_plugins_default', type: 'string', value: '', description: 'Default configuration option for main.plugins setting, used as fallback if the configuration option is unset')
-option('config_plugin_ibft', type: 'boolean', value: true, description: 'enable ibft configuration plugin')
+option('ibft',     type: 'boolean', value: false, description: 'enable ibft configuration plugin')
```
Comment (inigomartinez) [inline meson_options.txt]: There are a couple whitespaces in front of type that would be nice to remove.

File: libnm/meson.build [hunk 2]
```diff
@@ -19,6 +13,11 @@ cflags = [
   '-DNMRUNDIR="@0@"'.format(nm_pkgrundir)
 ]
 
+if have_fake_typelibs
+  deps += [ gir_dep ]
```
Comment (inigomartinez) [inline libnm/meson.build]: Square brackets are not necessary when adding a dependency to a list.

File: meson_options.txt (hunk 1 above)
Comment (thom311) [reply meson_options.txt]: the white space is intentional, to vertically align with the neighbouring lines, that do something similar.

File: libnm/meson.build (hunk 2 above)
Comment (thom311) [reply libnm/meson.build]: fixed

**Comment by thom311** [comment]
> Why not create each linker script strings in the main meson.build file and reuse them when needed?

done. See commit 34cb6f9877a9b7a7032899dbccca1def88da4191

File: libnm/meson.build (hunk 2 above)
Comment (thom311) [reply libnm/meson.build]: fixed by commit 98b46537fa1a91941b5f7a8252b3fce85554a855

**Pull request merged**